omod-2.4/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/search/openmrs2_4/OrderSearchHandler2_4.java

```diff
@@ -0,0 +1,223 @@
+		String activatedOnOrBeforeDateStr = context.getParameter("activatedOnOrBeforeDate");
+		String activatedOnOrAfterDateStr = context.getParameter("activatedOnOrAfterDate");
+		String isStoppedStr = context.getParameter(REQUEST_PARAM_IS_STOPPED);
+		String autoExpireOnOrBeforeDateStr = context.getParameter(REQUEST_PARAM_AUTO_EXPIRE_ON_OR_BEFORE_DATE);
+		String canceledOrExpiredOnOrBeforeDateStr = context
+		        .getParameter(REQUEST_PARAM_CANCELED_OR_AUTO_EXPIRE_ON_OR_BEFORE_DATE);
+		String actionStr = context.getParameter(REQUEST_PARAM_ACTION);
+		String fulfillerStatusStr = context.getParameter(REQUEST_PARAM_FULFILLER_STATUS);
+		String excludeCanceledAndExpiredStr = context.getParameter(REQUEST_PARAM_EXCLUDE_CANCELED_AND_EXPIRED);
+		String includeVoidedStr = context.getParameter("includeVoided");
+		
+		// build search criteria for order service
+		boolean includeVoided = StringUtils.isNotBlank(includeVoidedStr) ? Boolean.parseBoolean(includeVoidedStr) : false;
+		// by default the Canceled(dateStopped != null) and Expired(autoExpire < today) orders are excluded
+		boolean excludeCanceledAndExpired = StringUtils.isNotBlank(excludeCanceledAndExpiredStr) ? Boolean
+		        .parseBoolean(excludeCanceledAndExpiredStr) : false;
+		boolean isStopped = StringUtils.isNotBlank(isStoppedStr) ? Boolean.parseBoolean(isStoppedStr) : false;
+		
+		Date activatedOnOrBeforeDate = StringUtils.isNotBlank(activatedOnOrBeforeDateStr) ?
+		        (Date) ConversionUtil.convert(activatedOnOrBeforeDateStr, Date.class) : null;
+		Date activatedOnOrAfterDate = StringUtils.isNotBlank(activatedOnOrAfterDateStr) ?
+		        (Date) ConversionUtil.convert(activatedOnOrAfterDateStr, Date.class) : null;
+		Date autoExpireOnOrBeforeDate = StringUtils.isNotBlank(autoExpireOnOrBeforeDateStr) ?
+		        (Date) ConversionUtil.convert(autoExpireOnOrBeforeDateStr, Date.class) : null;
+		Date canceledOrExpiredOnOrBeforeDate = StringUtils.isNotBlank(canceledOrExpiredOnOrBeforeDateStr) ?
+		        (Date) ConversionUtil.convert(canceledOrExpiredOnOrBeforeDateStr, Date.class) : null;
+		Order.Action action = StringUtils.isNotBlank(actionStr) ? Order.Action
+		        .valueOf(actionStr) : null;
+		Order.FulfillerStatus fulfillerStatus = StringUtils.isNotBlank(fulfillerStatusStr) ? Order.FulfillerStatus
+		        .valueOf(fulfillerStatusStr) : null;
+		List<Concept> concepts = null;
+		List<OrderType> orderTypes = null;
+		
+		Patient patient = null;
+		if (StringUtils.isNotBlank(patientUuid)) {
+			patient = patientService.getPatientByUuid(patientUuid);
+			if (patient == null) {
+				throw new ObjectNotFoundException();
+			}
+		}
+		
+		CareSetting careSetting = null;
+		if (StringUtils.isNotBlank(careSettingUuid)) {
+			careSetting = orderService.getCareSettingByUuid(careSettingUuid);
+			if (careSetting == null) {
+				throw new ObjectNotFoundException();
+			}
+		}
+		
+		// if none of the uuids are found, throw an exception
```

Comment:
We probably want to throw an exception if *any* of the concepts aren't found? (And also if any of the order types aren't found?)